File: Cap04/python/maior_elemento.py
def maiorElemento(x):
    if len(x) == 1:
        return x[0]
    
    m = maiorElemento(x[1:])
    if m > x[0]:
        return m
    else:
        return x[0]
    
print(maiorElemento([1, 2, 3, 4, 5])) # 5

File: Cap04/golang/maiorElemento.go
package main

import "fmt"

func maiorElemento(x []int) int {
	if len(x) == 1 {
		return x[0]
	}

	m := maiorElemento(x[1:])
	if m > x[0] {
		return m
	}
	return x[0]
}

func main() {
	lista := []int{1, 2, 3, 4, 5}
	result := maiorElemento(lista)
	fmt.Println(result)
}
